File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>>res;
        unordered_map<string, vector<string>>mp;
        
        for(string s : strs){
            string t = s;
            sort(s.begin(), s.end());
            mp[s].push_back(t);
        }
        
        for(auto it : mp){
            res.push_back(it.second);
        }
        
        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>>res = new ArrayList<>();
        Map<String, List<String>>mp = new HashMap<>();
        
        for(String s : strs){
            String t = s;
            
            char temp[] = s.toCharArray();
            Arrays.sort(temp);
            String key = new String(temp);
            
            List<String>currList = mp.getOrDefault(key, new ArrayList<String>());
            
            currList.add(t);
            
            mp.put(key, currList);
        }
        
        for(List<String>list : mp.values()){
            res.add(list);
        }
        
        return res;
    }
}